tests/foreman/ui_airgun/test_ldap_authentication.py

```diff
@@ -241,3 +313,371 @@ def test_positive_create_with_idm_org_and_loc(session, ipa_data):
+        session.usergroup.refresh_external_group(ldap_usergroup_name, EXTERNAL_GROUP_NAME)
+    with Session(
+            test_name,
+            ldap_data['ldap_user_name'],
+            ldap_data['ldap_user_passwd'],
+    ) as session:
+        if bz_bug_is_open(1652938):
+            try:
+                session.activationkey.search('')
+            except NoSuchElementException:
+                session.browser.refresh()
+        session.activationkey.create({'name': ak_name})
+        assert session.activationkey.search(ak_name)[0]['Name'] == ak_name
+        current_user = session.activationkey.read(ak_name, 'current_user')['current_user']
+        assert current_user == ldap_data['ldap_user_name']
+
+
+@upgrade
+@tier2
+def test_positive_update_external_roles(
+        session, ldap_data, ldap_user_name, test_name, auth_source, ldap_usergroup_name):
+    """Added AD UserGroup roles get pushed down to user
+
+    :id: f3ca1aae-5461-4af3-a508-82679bb6afed
+
+    :setup: assign additional roles to the UserGroup
+
+    :steps:
+        1. Create an UserGroup.
+        2. Assign some roles to UserGroup.
+        3. Create an External AD UserGroup as per the UserGroup name in AD.
+        4. Login to sat6 with the AD user.
+        5. Assign additional roles to the UserGroup.
+        6. Login to sat6 with LDAP user that is part of aforementioned
+           UserGroup.
+
+    :expectedresults: User has access to all NEW functional areas that are
+        assigned to aforementioned UserGroup.
+
+    :CaseLevel: Integration
+    """
+    ak_name = gen_string('alpha')
+    auth_source_name = 'LDAP-' + auth_source.name
+    location_name = gen_string('alpha')
+    foreman_role = entities.Role().create()
+    katello_role = entities.Role().create()
+    foreman_permissions = {'Location': PERMISSIONS['Location']}
+    katello_permissions = {'Katello::ActivationKey': PERMISSIONS['Katello::ActivationKey']}
+    create_role_permissions(foreman_role, foreman_permissions)
+    create_role_permissions(katello_role, katello_permissions)
+    with session:
+        session.usergroup.create({
+            'usergroup.name': ldap_usergroup_name,
+            'roles.resources.assigned': [foreman_role.name],
+            'external_groups.name': EXTERNAL_GROUP_NAME,
+            'external_groups.auth_source': auth_source_name,
+        })
+        assert session.usergroup.search(ldap_usergroup_name)[0]['Name'] == ldap_usergroup_name
+        session.user.update(ldap_data['ldap_user_name'], {'user.auth': auth_source_name})
+        with Session(
+                test_name,
+                ldap_data['ldap_user_name'],
+                ldap_data['ldap_user_passwd'],
+        ) as ldapsession:
+            ldapsession.location.create({'name': location_name})
+            assert ldapsession.location.search(location_name)[0]['Name'] == location_name
+            current_user = ldapsession.location.read(location_name, 'current_user')['current_user']
```

Comment:
just a question =)
Location entity/view doesn't have widget like `current_user`, so you get that data from common view. Is it possible to read that data omitting navigation to location All->Current views (so basically whole redundant navigation will be skipped)?

Reply:
I'd like to, but we don't have access to views directly from test layer (or rather shouldn't access them). But there won't be a big overkill since previous command was location search, so basically we only do one extra click and then read user.